app/services/page_content.rb

```diff
@@ -0,0 +1,43 @@
+class PageContent
+  attr_accessor :book_id, :page_id, :content, :doc
+
+  def initialize(book_uuid:, book_version:, page_uuid:)
+    @book_id = "#{book_uuid}@#{book_version}"
+    @page_id = "#{book_id}:#{page_uuid}"
+  end
+
+  def s3
+    @s3 ||= OpenStax::Content::S3.new
+  end
+
+  def latest_archive_version
+    @latest_archive_version ||= s3.ls.last
+  end
+
+  def archive
+    @archive ||= OpenStax::Content::Archive.new latest_archive_version
```

Comment:
you should use the archive that rex uses, https://openstax.org/rex/config.json

technically each book could also have an archive override https://staging.openstax.org/rex/release.json

there is a chance if you don't use the same book that your desired book/version will not be found

its still a little janky because it won't be environmentally aware, i'm not sure how to do that without adding more config to the api service. its probably fine

Reply:
Updated to use a secrets `archive_version` setting: https://github.com/openstax/highlights-api/pull/79/commits/9a3cdd9ea3ce0f928478814afc57f28283d1eb0e

I didn't know that about the book override, interesting! Now I wonder if that's fetchable... If so I could use it when initializing the content fetch.

Reply:
what do you mean fetchable?